### PR TITLE
[iDEAL] Bank Selection Screen

### PR DESCRIPTION
**Pull request by mpetrenco**
- Added custom iDEAL bank UITableViewCell;
- Added bank selection UITableViewController;
- Added bank icons;

**Preview:**
Screen Shot 2019-10-16 at 1 43 28 PM

### CODE REVIEW

File: Source/iDEAL/Cells/IDEALBankTableViewCell.m [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+@property (weak, nonatomic) IBOutlet UIImageView *bankLogoImageView;
+
+@end
+
+@implementation IDEALBankTableViewCell
+
+- (void)configureWithBank:(IDEALBank *)bank {
+    NSString *iconName = [NSString stringWithFormat:@"logo-%@", bank.bankIdentifierCode];
+    self.bankLogoImageView.image =[UIImage imageNamed:iconName];
```
Comment (zfoltin) [inline Source/iDEAL/Cells/IDEALBankTableViewCell.m]: Will the merchant app need to copy over these images to work? Can we make a resource bundle with these images? or at least https://stackoverflow.com/a/28777893/916656

File: Source/iDEAL/Cells/IDEALBankTableViewCell.xib [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        <placeholder placeholderIdentifier="IBFilesOwner" id="-1" userLabel="File's Owner"/>
+        <placeholder placeholderIdentifier="IBFirstResponder" id="-2" customClass="UIResponder"/>
+        <tableViewCell contentMode="scaleToFill" selectionStyle="default" indentationWidth="10" reuseIdentifier="IDEALBankTableViewCell" rowHeight="73" id="KGk-i7-Jjw" customClass="IDEALBankTableViewCell">
+            <rect key="frame" x="0.0" y="0.0" width="320" height="73"/>
+            <autoresizingMask key="autoresizingMask" flexibleMaxX="YES" flexibleMaxY="YES"/>
+            <tableViewCellContentView key="contentView" opaque="NO" clipsSubviews="YES" multipleTouchEnabled="YES" contentMode="center" tableViewCell="KGk-i7-Jjw" id="H2p-sc-9uM">
+                <rect key="frame" x="0.0" y="0.0" width="320" height="73"/>
+                <autoresizingMask key="autoresizingMask"/>
+                <subviews>
+                    <imageView clipsSubviews="YES" userInteractionEnabled="NO" contentMode="scaleAspectFit" horizontalHuggingPriority="251" verticalHuggingPriority="251" translatesAutoresizingMaskIntoConstraints="NO" id="gcg-oo-R5r">
```
Comment (zfoltin) [inline Source/iDEAL/Cells/IDEALBankTableViewCell.xib]: How does this work with VoiceOver? Let's try and make it as accessible as possible

File: Source/iDEAL/IDEALBankSelectionTableViewController.m [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+//  SOFTWARE.
+
+#import "IDEALBankSelectionTableViewController.h"
+#import "IDEALBankTableViewCell.h"
+#import "IDEALBank.h"
+
+@implementation IDEALBankSelectionTableViewController
+
+# pragma mark - Private properties
```
Comment (zfoltin) [inline Source/iDEAL/IDEALBankSelectionTableViewController.m]: I'm not a big fan of `# pragma mark`, especially when the whole class fits on two screens.

File: Source/iDEAL/IDEALBankSelectionTableViewController.m [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+
+- (NSString *)bankCellIdentifier {
```
Comment (zfoltin) [inline Source/iDEAL/IDEALBankSelectionTableViewController.m]: Does this need to be a function? How about a simple const?

File: JudoKitObjCExampleApp/Assets.xcassets/ideal-logos/logo-ABNANL2A.imageset/Contents.json [hunk 5]
```diff
@@ -0,0 +1,21 @@
+{
+  "images" : [
+    {
+      "idiom" : "universal",
+      "filename" : "logo-ABNANL2A.png",
```
[the remark this answers is not in the document]
Comment (mpetrenco) [reply JudoKitObjCExampleApp/Assets.xcassets/ideal-logos/logo-ABNANL2A.imageset/Contents.json]: Sorry about that, Zeno.
These were the images that were provided by Jumaane. 
I'll ask him if he has access to PDF logos.

File: Source/iDEAL/IDEALBankSelectionTableViewController.m (hunk 3 above)
Comment (mpetrenco) [reply Source/iDEAL/IDEALBankSelectionTableViewController.m]: Removed the `# pragma mark` comments

File: Source/iDEAL/IDEALBankSelectionTableViewController.m (hunk 4 above)
Comment (mpetrenco) [reply Source/iDEAL/IDEALBankSelectionTableViewController.m]: Converted into a constant

File: Source/iDEAL/Cells/IDEALBankTableViewCell.xib (hunk 2 above)
Comment (mpetrenco) [reply Source/iDEAL/Cells/IDEALBankTableViewCell.xib]: Added accessibility label and hint

File: JudoKitObjCExampleApp/Assets.xcassets/ideal-logos/logo-ABNANL2A.imageset/Contents.json (hunk 5 above)
[the remark this answers is not in the document]
Comment (mpetrenco) [reply JudoKitObjCExampleApp/Assets.xcassets/ideal-logos/logo-ABNANL2A.imageset/Contents.json]: Jumaane replied that he does not have access to either vector images or higher resolution PNGs.
I will keep these for now.

File: Source/iDEAL/Cells/IDEALBankTableViewCell.m (hunk 1 above)
Comment (mpetrenco) [reply Source/iDEAL/Cells/IDEALBankTableViewCell.m]: Added the iDEAL logos as part of the `icons.bundle` resource file

File: Source/iDEAL/Cells/IDEALBankTableViewCell.xib (hunk 2 above)
Comment (stpopa) [reply Source/iDEAL/Cells/IDEALBankTableViewCell.xib]: > How does this work with VoiceOver? Let's try and make it as accessible as possible

This is out of our current sprint scope

File: Source/iDEAL/Cells/IDEALBankTableViewCell.m [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+
+@property (weak, nonatomic) IBOutlet UIImageView *bankLogoImageView;
+
+@end
+
+@implementation IDEALBankTableViewCell
+
+- (void)configureWithBank:(IDEALBank *)bank {
+    
+    NSBundle *bundle = [NSBundle bundleForClass:IDEALBankTableViewCell.class];
```
Comment (stpopa) [inline Source/iDEAL/Cells/IDEALBankTableViewCell.m]: Let's extract the logic of retrieving the icons file paths to a category, we use this in 2 places

File: Source/iDEAL/IDEALBankSelectionTableViewController.m [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
+    return self.bankList.count;
+}
+
+- (UITableViewCell *)tableView:(UITableView *)tableView
+         cellForRowAtIndexPath:(NSIndexPath *)indexPath {
+
+    IDEALBank *bank = self.bankList[indexPath.row];
+    
+    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:bankCellIdentifier
```
Comment (stpopa) [inline Source/iDEAL/IDEALBankSelectionTableViewController.m]: `IDEALBankTableViewCell *cell = (IDEALBankTableViewCell* ) [tableView dequeueReusableCellWithIdentifier:bankCellIdentifier forIndexPath:indexPath];` ?

**Pull request merged**